game-backend/src/MainController.ts

```diff
@@ -41,6 +44,32 @@ export default class MainController {
     }
   }
 
+  /**
+   * Looks up the connection id in the store.
+   * It returns the found connectionId or generates a new one respectively.
+   * @param requestedConnectionId The connection id the client requests
+   */
+  private getConnectionId(requestedConnectionId: string): ConnectionId {
+    if (
+      requestedConnectionId &&
+      this.store.userContexts.userContextIds.indexOf(requestedConnectionId)
+    ) {
+      return requestedConnectionId;
+    }
+    return this.hyperIdInstance();
+  }
+
+  private static sendConnectionId(ws: WebSocket, connectionId: string): void {
+    ws.send(
+      JSON.stringify({
+        type: 'connectionInitialization',
+        payload: {
+          connectionId,
+        },
+      } as ConnectionInitialization)
```

Comment:
Try not to do type casting - doing it using `as` will erase the benefit of using TypeScript.

For a reference implementation see:
https://github.com/code-and-conquer/darwin/blob/ae21189a07d11a4592aff2776b90d0ac64a03d72/game-frontend/src/components/user-script/index.tsx#L19

Reply:
Thank you, somehow I always mess that up...